pkg/util/net/firewall_dialer_test.go

```diff
@@ -0,0 +1,132 @@
+package net
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/cortexproject/cortex/pkg/util/flagext"
+)
+
+func TestFirewallDialer(t *testing.T) {
+	blockedCIDR := flagext.CIDR{}
+	require.NoError(t, blockedCIDR.Set("172.217.168.64/28"))
+
+	type testCase struct {
+		network       string
+		address       string
+		expectBlocked bool
+	}
+
+	tests := map[string]struct {
+		cfg   FirewallDialerConfig
+		cases []testCase
+	}{
+		"should not block traffic with default config": {
+			cfg: FirewallDialerConfig{},
+			cases: []testCase{
+				{"tcp", "localhost", false},
+				{"tcp", "127.0.0.1", false},
+				{"tcp", "google.com", false},
+				{"tcp", "172.217.168.78", false},
+			},
+		},
+		"should support blocking private addresses": {
+			cfg: FirewallDialerConfig{
+				BlockPrivate: true,
+			},
+			cases: []testCase{
+				{"tcp", "localhost", true},
+				{"tcp", "127.0.0.1", true},
+				{"tcp", "192.168.0.1", true},
+				{"tcp", "10.0.0.1", true},
+				{"tcp", "google.com", false},
+				{"tcp", "172.217.168.78", false},
+				{"tcp6", "fdf8:f53b:82e4::53", true},       // Local
```

Comment:
Not 100% sure, but I think in the common case the dialer should always be called with "tcp"